File: product-service/src/main/java/com/mealmap/productservice/core/dto/product/ProductCreatingDto.java
package com.mealmap.productservice.core.dto.product;

import com.mealmap.productservice.core.dto.nutrient.NutrientDto;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import lombok.Value;

import java.util.Set;

@Value
@Builder
public class ProductCreatingDto {
    @NotEmpty
    @Size(max = 40)
    String name;

    @NotNull
    @Positive
    Integer price;

    @NotNull
    @Positive
    Integer weight;

    @NotNull
    @Valid
    NutrientDto nutrients;

    @Size(max = 255)
    String description;

    @NotNull
    Integer supplierId;

    @NotEmpty
    Set<Long> categories;
}

File: frontend/src/api/product/dto/ProductCreatingDto.ts
import { NutrientDto } from "./NutrientDto";

export interface ProductCreatingDto {
  name: string; // Max length: 40
  imageUrl: string;
  price: number; // Must be positive
  weight: number; // Must be positive
  nutrients: NutrientDto; // Nutritional information
  description?: string; // Optional, max length: 255
  supplierId: number; // Must not be null
  categories: number[]; // List of category IDs
}
